lucet-runtime/lucet-runtime-internals/src/instance/signals.rs

```diff
@@ -218,8 +217,12 @@ extern "C" fn handle_signal(signum: c_int, siginfo_ptr: *mut siginfo_t, ucontext
                     // If the trap table lookup returned unknown, it is a fatal error
                     let unknown_fault = trapcode.is_none();
 
-                    // If the trap was a segv or bus fault and the addressed memory was in the
-                    // signal stack guard page or outside the alloc entirely, the fault is fatal
+                    // The stack of where we want to resume, a Lucet guest, may have been
+                    // exhausted. So, swap to an emergency stack to ensure we'll have at least a
+                    // few words of stack space as may be necessary for `Context::set_from_signal`
+                    // in debug builds.
+                    ctx.set_sp(inst.alloc.slot().emergency_stack_base());
```

Comment:
Will we ever end up at this point with something already on the emergency stack? My thoughts is "no" because any signals that arrive while doing the emergency work will be native code, and so we'd want to bring down the process anyway. But I want to make sure, and note the reasoning in the code so folks aren't worried about clobbering.